seacrowd/sea_datasets/prdect_id/prdect_id.py

```diff
@@ -0,0 +1,154 @@
+product review is annotated with a single emotion, i.e., love, happiness, anger,
+fear, or sadness. The group of annotators does the annotation process to provide
+emotion labels by following the emotions annotation criteria created by an
+expert in clinical psychology. Other attributes related to the product review
+are also extracted, such as Location, Price, Overall Rating, Number Sold, Total
+Review, and Customer Rating, to support further research.
+"""
+
+_HOMEPAGE = "https://data.mendeley.com/datasets/574v66hf2v/1"
+_LICENSE = Licenses.CC_BY_4_0.value
+_URL = "https://data.mendeley.com/public-files/datasets/574v66hf2v/files/f258d159-c678-42f1-9634-edf091a0b1f3/file_downloaded"
+
+_SUPPORTED_TASKS = [Tasks.SENTIMENT_ANALYSIS, Tasks.EMOTION_CLASSIFICATION]
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+class PrdectIDDataset(datasets.GeneratorBasedBuilder):
+    """PRDECT-ID Dataset"""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    SEACROWD_SCHEMA_NAME = "text"
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=_DATASETNAME,
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_emotion_seacrowd_{SEACROWD_SCHEMA_NAME}",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema for emotion classification",
+            schema=f"seacrowd_{SEACROWD_SCHEMA_NAME}",
+            subset_id=f"{_DATASETNAME}_emotion",
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_sentiment_seacrowd_{SEACROWD_SCHEMA_NAME}",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema for sentiment analysis",
+            schema=f"seacrowd_{SEACROWD_SCHEMA_NAME}",
+            subset_id=f"{_DATASETNAME}_sentiment",
+        )
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+    CLASS_LABELS_EMOTION = ["Happy", "Sadness", "Anger", "Love", "Fear"]
+    CLASS_LABELS_SENTIMENT = ["Positive", "Negative"]
+
+    def _info(self) -> datasets.DatasetInfo:
+        task = self.config.subset_id.split("_")[-1]
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "Category": datasets.Value("string"),	
+                    "Product Name": datasets.Value("string"),	
+                    "Location": datasets.Value("string"),	
+                    "Price": datasets.Value("int32"),
+                    "Overall Rating": datasets.Value("float32"),	
+                    "Number Sold": datasets.Value("int32"),	
+                    "Total Review": datasets.Value("int32"),	
+                    "Customer Rating": datasets.Value("string"),
```

Comment:
why use string instead of int?

Reply:
Ah `int` looks more correct. Will change